src/main/scala/workflow/graph/Transformer.scala

```diff
@@ -15,26 +15,17 @@ import scala.reflect.ClassTag
  * @tparam A input item type the transformer takes
  * @tparam B output item type the transformer produces
  */
-abstract class Transformer[A, B : ClassTag] extends TransformerOperator with Pipeline[A, B] {
-  /**
-   * The executor this transformer contains under the hood (because it extends Pipeline)
-   */
-  @transient override private[graph] lazy val executor = new GraphExecutor(Graph(
-    sources = Set(SourceId(0)),
-    sinkDependencies = Map(SinkId(0) -> NodeId(0)),
-    operators = Map(NodeId(0) -> this),
-    dependencies = Map(NodeId(0) -> Seq(SourceId(0)))
-  ))
-
-  /**
-   * The source this transformer contains under the hood (because it extends Pipeline)
-   */
-  override private[graph] val source = SourceId(0)
-
-  /**
-   * The sink this transformer contains under the hood (because it extends Pipeline)
-   */
-  override private[graph] val sink = SinkId(0)
+abstract class Transformer[A, B : ClassTag] extends TransformerOperator with Chainable[A, B] {
+  private[graph] override def toPipeline: Pipeline[A, B] = new Pipeline(
```

Comment:
So if I implement `Chainable` I have to know how to convert myself to a pipeline - is that the right interpretation? And that's it?